raster-foundry/raster-foundry
# Allow object resharing on the frontend

## Overview
Use object permissions to determine whether a user can share on the frontend

Allows sharing on:
* Projects
* Analyses
* Templates
* Scenes

Added ownership filters, but can't fully share yet due to concerns requiring further investigation:
* Shapes
* Datasources

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo
![image](https://user-images.githubusercontent.com/4392704/51621770-3705ca00-1f03-11e9-823b-c206f98f1043.png)
![image](https://user-images.githubusercontent.com/4392704/51621798-47b64000-1f03-11e9-8ec6-a6f33e616e46.png)
![image](https://user-images.githubusercontent.com/4392704/51621810-4dac2100-1f03-11e9-9b12-b9e3c6f5c770.png)
![image](https://user-images.githubusercontent.com/4392704/51621821-543a9880-1f03-11e9-87e6-9b8a649ba41d.png)
![image](https://user-images.githubusercontent.com/4392704/51621824-5997e300-1f03-11e9-81c1-5e5a08227b2f.png)
![image](https://user-images.githubusercontent.com/4392704/51621842-60bef100-1f03-11e9-9d0f-efb039128210.png)
![image](https://user-images.githubusercontent.com/4392704/51621854-69172c00-1f03-11e9-8a14-bfafb7930645.png)


### Notes
Leaving shapes and datasources to another issue because they require some thinking about

## Testing Instructions

* Using two separate accounts, Verify that the listed objects are shareable
    * Projects
	* Analyses
	* Templates
	* Scenes
* Verify that the ownership filters apply correctly


Closes #4489
